tools/release-posts/commands/release-post/release-post-contributors.ts

```diff
@@ -28,9 +29,37 @@ const program = new Command()
 	.action( async ( currentVersion, previousVersion ) => {
 		Logger.startTask( 'Generating contributors list...' );
 
+		const currentParsed = semver.parse( currentVersion );
+		const previousParsed = semver.parse( previousVersion );
+		let currentBranch;
+		let previousBranch;
+		
+		try {
+			if ( ! currentParsed ) {
+				throw new Error( 'Unable to parse current version' );
+			}
+			currentBranch = `release/${ currentParsed.major }.${ currentParsed.minor }`;
+		} catch ( error: unknown ) {
+			Logger.notice(
+				`Unable to find '${ currentBranch }', using 'trunk'.`
+			);
+			currentBranch = 'trunk';
+		}
+
+		try {
+			if ( ! previousParsed ) {
+				throw new Error( 'Unable to parse previous version' );
+			}
+			previousBranch = `release/${ previousParsed.major }.${ previousParsed.minor }`;
+		} catch ( error: unknown ) {
+			throw new Error(
+				`Unable to find '${ previousBranch }'. Branch for previous version must exist.`
+			);
+		}
+
```

Comment:
Neither of these should be try/catch, it increases confusion in the code paths here (much harder to quickly tell what's happening since you keep throwing and catching as a way to handle different code paths). You don't need try/catch unless you're calling APIs that by behaviour will throw exceptions in some states.

Instead you should just check these conditions and when you want to crash the CLI call `Logger.error` which will log and exit.
```suggestion

	if (  ! currentParsed ) {
		Logger.notice( 'Unable to parse current version' );
		currentBranch = 'trunk';
	} else {
	    currentBranch = `release/${ currentParsed.major }.${ currentParsed.minor }`;
	}
	
	if ( ! previousParsed ) {
	  // Logger error does process.exit and logs the message
	  Logger.error(`Unable to find '${ previousBranch }'. Branch for previous version must exist.`)
	} else {
	  previousBranch = `release/${ previousParsed.major }.${ previousParsed.minor }`;
	}


```

Reply:
@samueljseay looking at it now I do agree that the exception strategy here seems a bit odd. I took it from `release-post-release.ts`. I'd like to accept your suggestion and then rework things in `release-post-release.ts` in a similar manner as well, but before doing that I'd like to know whether @jonathansadowski maybe had a specific rationale for writing it this way?